lib/archethic/contracts/interpreter/utils.ex

```diff
@@ -215,9 +217,9 @@ defmodule Archethic.Contracts.Interpreter.Utils do
     {node, acc}
   end
 
-  # Whitelist the hash/1 function
-  def prewalk(node = {{:atom, "hash"}, _, [_data]}, acc = {:ok, %{scope: scope}})
-      when scope != :root,
+  # Whitelist the hash/2 function
+  def prewalk(node = {{:atom, "hash"}, _, [_data, algo]}, acc = {:ok, %{scope: scope}})
```

Comment:
I think you should keep `hash/1` to be backward compatible and use the default hash algorithm in the library function

Reply:
done